### PR TITLE
Feat: upgrade to go version 1.18

### LINKED ISSUE #336: Upgrade to go version 1.18

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #336 

### Solution
Located all the 1.16 locations and modified them to 1.18.
Ran: `go mod tidy`.
(Unrelated) Formatted some terraform file (`terraform fmt`).

### CODE REVIEW

**Comment by TomerHeber** [comment]
@yaronya - there's a comment in the code `// please change also in repo secrets`. Do you know what's it about?

**Comment by yaronya** [comment]
> @yaronya - there's a comment in the code `// please change also in repo secrets`. Do you know what's it about?

@TomerHeber I think we used to fetch the version as a secret from GH (since that's the only way to share a common value between multiple GH actions) but we changed it back and only comment was left. You can ignore it probably.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
       - name: Go staticcheck
         uses: dominikh/staticcheck-action@v1.2.0
         with:
-          version: "2021.1.2"
+          version: "2022.1"
```
Comment (TomerHeber) [inline .github/workflows/ci.yml:39]: Go 1.18 requires a newer staticcheck.

**Review by razbensimon — APPROVED** [review]
LGTM

**Comment by razbensimon** [comment]
> > @yaronya - there's a comment in the code `// please change also in repo secrets`. Do you know what's it about?
> 
> @TomerHeber I think we used to fetch the version as a secret from GH (since that's the only way to share a common value between multiple GH actions) but we changed it back and only comment was left. You can ignore it probably.

So maybe let's remove that comment? @yaronya 
@TomerHeber

**Pull request merged**